.clangd.example

```diff
@@ -0,0 +1,35 @@
+# Global config for all files
+CompileFlags:
+  Add: [
+    # Remove all built-in definitions
+    "-undef",
+
+    # Indicator for things which shouldn't be processed in IDEs
+    "-DDECOMP_IDE_FLAG",
+
+    # No standard includes
+    "-nostdinc",
+    "-nostdinc++",
+    "-fno-builtin",
+
+    # Target definitions
+    "--target=ppc32-unknown-eabi",
+  ]
+---
+# C files
+If:
+  PathMatch: .*\.c
+CompileFlags:
+  Add: [
+    "--language=c",
+    "-std=c99",
+  ]
+---
+# C++ files
+If:
+  PathMatch: .*\.cpp?
+CompileFlags:
+  Add: [
+    "--language=c++",
+    "-std=c++03",
+  ]
```

Comment:
Aren't we adding the language flag already?

Reply:
Currently it depends on if the object has the appropriate cflags in its configuration. If a project doesn't add e.g. `-lang=c` or `-lang=c++`, it won't emit these flags into compile_commands.json.

Probably would be good to do this in the generation itself, that way we can add appropriate defaults based on the extension if one isn't already specified.

Reply:
Oh, are you referring to the objdiff config generation? I didn't notice that added the language flag already lol

Reply:
Ah, yeah, maybe we can reuse that logic in both places

Reply:
Moved that to `generate_build_ninja`, and applied it directly to the object config so that it can flow to other generation tasks properly.

Reply:
Can we remove this from .clangd then? Or is there still a purpose?

Reply:
We can, yeah, forgot to do so after making the change.